post9p.go

```diff
@@ -11,23 +11,16 @@ import (
 
 var chattyfuse bool
 
-func post9pservice(fd *os.File, name string, mtpt string) int {
-	var (
-		err      error
-		ns, addr string
-		conn     *net.UnixConn
-	)
-
+func post9pservice(conn net.Conn, name string, mtpt string) int {
```

Comment:
I feel that this code might be nicer if we had the plan9, darwin, linux (and eventually windows apparently) versions were in separate files. WYDT?

Conceivably in a different CL I suppose.

Reply:
A portable version is easier to maintain and test. Windows and Plan 9 doesn't have `syscall.Socketpair`. [`net.Pipe`](https://golang.org/src/net/pipe.go) is implemented in a very portable way -- it doesn't even call anything in `syscall` package. It just passes buffers around in channels.

Plan 9 will of course need a separate implementation in the future because we actually need a file descriptor that needs to be posted to `/srv`. `net.Pipe` doesn't give you a file descriptor.

Reply:
I agree with you about portable code. All code that can be portable should be. I was more commenting on the mounting logic. There is code to launch fuse or use the Linux mount command. On Plan9, this would be a call to `mount` I presume. This code I think could be separated out into separate per-platform files.